modules/datastore/src/Form/DashboardForm.php

```diff
@@ -0,0 +1,437 @@
+   *
+   * @var \Drupal\Core\Pager\PagerManagerInterface
+   */
+  protected $pagerManager;
+
+  /**
+   * Items per page.
+   *
+   * @var int
+   */
+  protected $itemsPerPage;
+
+  /**
+   * DashboardController constructor.
+   *
+   * @param \Drupal\harvest\Service $harvestService
+   *   Harvest service.
+   * @param \Drupal\common\DatasetInfo $datasetInfo
+   *   Dataset information service.
+   * @param \Drupal\metastore\Service $metastoreService
+   *   Metastore service.
+   * @param \Drupal\Core\Pager\PagerManagerInterface $pagerManager
+   *   Pager manager service.
+   */
+  public function __construct(
+    Service $harvestService,
+    DatasetInfo $datasetInfo,
+    MetastoreService $metastoreService,
+    PagerManagerInterface $pagerManager
+  ) {
+    $this->harvest = $harvestService;
+    $this->datasetInfo = $datasetInfo;
+    $this->metastore = $metastoreService;
+    $this->pagerManager = $pagerManager;
+    $this->itemsPerPage = 10;
+  }
+
+  /**
+   * Create controller object from dependency injection container.
+   */
+  public static function create(ContainerInterface $container): self {
+    return new static(
+      $container->get('dkan.harvest.service'),
+      $container->get('dkan.common.dataset_info'),
+      $container->get('dkan.metastore.service'),
+      $container->get('pager.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'dashboard_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+    // Set the method.
+    $form_state->setMethod('GET');
+    // Fetch GET parameter.
+    $params = $this->getParameters();
+    // Add custom after_build method to remove unnecessary GET parameters.
+    $form['#after_build'] = ['::afterBuild'];
+
+    // Build dataset import status table render array.
+    return $form + $this->buildFilters($params) + $this->buildTable($this->getDatasets($params));
+  }
+
+  /**
+   * Fetch request GET parameters.
+   *
+   * @return array
+   *   Request GET parameters.
+   */
+  protected function getParameters(): array {
+    return ($request = $this->getRequest()) && isset($request->query) ? array_filter($request->query->all()) : [];
+  }
+
+  /**
+   * Custom after build callback method.
+   */
+  public function afterBuild(array $element, FormStateInterface $form_state): array {
+    // Remove the form_token, form_build_id, form_id, and op from the GET
+    // parameters.
+    unset($element['form_token'], $element['form_build_id'], $element['form_id'], $element['filters']['actions']['submit']['#name']);
+
+    return $element;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {}
+
+  /**
+   * Build datasets import status table filters.
+   *
+   * @param string[] $filters
+   *   Dataset filters.
+   *
+   * @return array[]
+   *   Table filters render array.
+   */
```

Comment:
```suggestion
   *
   * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
   */
```

Reply:
hmm neat trick. ugly but effective!

Reply:
heh nevermind, does not look like CC catches that

Reply:
Yeah, I've tried everything I can think of. I thought this particular warning was coming from PHPMD, but I guess not...

> Let's override codeclimate on the buildFilters() method -- sometimes Drupal's just gonna Drupal...

How I can do this?